test_conformance/api/test_clone_kernel.cpp

```diff
@@ -351,51 +466,468 @@ int test_clone_kernel(cl_device_id deviceID, cl_context context, cl_command_queu
+                                            "set_kernel_exec_info_kernel");
+        test_error(error, "Unable to create srcKernel");
+
+        BufPtr* pBuf =
+            (BufPtr*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(BufPtr), 0);
+        cl_int* svmPtr_srcKernel =
+            (cl_int*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(cl_int), 0);
+        cl_int* svmPtr_srcKernel_1 =
+            (cl_int*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(cl_int), 0);
+        cl_int* svmPtr_cloneKernel_1 =
+            (cl_int*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(cl_int), 0);
+        cl_int* svmPtr_cloneKernel_2 =
+            (cl_int*)clSVMAlloc(context, CL_MEM_READ_WRITE, sizeof(cl_int), 0);
+        test_assert_error(pBuf != NULL || svmPtr_srcKernel != NULL
+                              || svmPtr_cloneKernel_1 != NULL
+                              || svmPtr_srcKernel_1 != NULL
+                              || svmPtr_cloneKernel_2 != NULL,
+                          "clSVMAlloc returned NULL");
+
+        // srcKernel, set args
+        if (test_svm_exec_info_helper(context, queue, pBuf, svmPtr_srcKernel,
+                                      &srcKernel, &intargs[0])
+            != 0)
+        {
+            test_fail("test_svm_exec_info_helper failed for srcKernel.\n");
+        }
+        clSVMFree(context, svmPtr_srcKernel);
+
+        // clone the srcKernel and set args
+        clKernelWrapper cloneKernel_1 = clCloneKernel(srcKernel, &error);
+        test_error(error, "clCloneKernel failed for cloneKernel_1");
+        if (test_svm_exec_info_helper(context, queue, pBuf,
+                                      svmPtr_cloneKernel_1, &cloneKernel_1,
+                                      &intargs[1])
+            != 0)
+        {
+            test_fail("test_svm_exec_info_helper failed for cloneKernel_1.\n");
+        }
+
+        // clone the cloneKernel_1 and set args
+        clKernelWrapper cloneKernel_2 = clCloneKernel(cloneKernel_1, &error);
+        test_error(error, "clCloneKernel failed for cloneKernel_2");
+        if (test_svm_exec_info_helper(context, queue, pBuf,
+                                      svmPtr_cloneKernel_2, &cloneKernel_2,
+                                      &intargs[2])
+            != 0)
+        {
+            test_fail("test_svm_exec_info_helper failed for cloneKernel_2.\n");
+        }
+
+        // enqueue - srcKernel again with different svm_ptr and args
+        if (test_svm_exec_info_helper(context, queue, pBuf, svmPtr_srcKernel_1,
+                                      &srcKernel, &intargs[3])
+            != 0)
+        {
+            test_fail("test_svm_exec_info_helper failed for srcKernel with "
+                      "different values.\n");
+        }
+        clSVMFree(context, svmPtr_srcKernel_1);
+
+        // enqueue - cloneKernel_1 again, to check if the args were not modified
+        if (test_exec_enqueue_helper(context, queue, pBuf, svmPtr_cloneKernel_1,
```

Comment:
From this point on, `pBuf` still contains pointer to `svmPtr_srcKernel_1`.
Since `test_exec_enqueue_helper` does not update the `pBuf`, the `cloneKernel_1` will receive an invalid pointer as argument.

The pointer inside `pBuf` needs to be updated before enqueuing the kernel.

Reply:
I have moved `clSVMFree` to free the buffers in the end.